reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -6162,9 +6216,39 @@ public int getPrefetch() {
 	 */
 	public final Flux<T> mergeOrderedWith(Publisher<? extends T> other,
 			Comparator<? super T> otherComparator) {
+		return mergeOrderedWith(other, otherComparator, true);
+	}
+
+	/**
+	 * Merge data from this {@link Flux} and a {@link Publisher} into a reordered merge
+	 * sequence, by picking the smallest value from each sequence as defined by a provided
+	 * {@link Comparator}. Note that subsequent calls are combined, and their comparators are
+	 * in lexicographic order as defined by {@link Comparator#thenComparing(Comparator)}.
+	 * <p>
+	 * The combination step is avoided if the two {@link Comparator Comparators} are
+	 * {@link Comparator#equals(Object) equal} (which can easily be achieved by using the
+	 * same reference, and is also always true of {@link Comparator#naturalOrder()}).
+	 * <p>
+	 * Note that merge is tailored to work with asynchronous sources or finite sources. When dealing with
+	 * an infinite source that doesn't already publish on a dedicated Scheduler, you must isolate that source
+	 * in its own Scheduler, as merge would otherwise attempt to drain it before subscribing to
+	 * another source.
+	 * <p>
+	 * <img class="marble" src="doc-files/marbles/mergeOrderedWith.svg" alt="">
+	 *
+	 * @param other the {@link Publisher} to merge with
+	 * @param otherComparator the {@link Comparator} to use for merging
+	 * @param delayError delaying errors until all data is consumed
+	 *
+	 * @return a new {@link Flux}
+	 */
+	public final Flux<T> mergeOrderedWith(Publisher<? extends T> other,
```

Comment:
looking at other merge*With operators, none of these differenciate on delayError. instead, the `mergeAdditionalSource` takes the `delayError` parameter of the first instance.

so if you _want_ to delay errors, you'd do eg. `Flux.mergeDelayErrors(a, b).mergeWith(c)` and the resulting `Flux` will delay errors from a, b and c.

we should apply the same logic here: only offer a `mergeOrderedWith` API with "default" behavior.

Reply:
Yes I can change that, I was just not sure if it would be readable enough, because one can build the pipeline using utility function so does not need to see "around". You can have input Flux of mergeOrdered(A, B) to some utility and then just do mergeOrderedWith (C). That second merge will be then inheriting from input which does not need to be expected.

Also you can just do input.mergeOrderedWith(A), then we will need mergeOrderedWithDelayErrors as well as there is nothing to inherit.